### PR TITLE
[Doc] Document LSTM is_init after slice sampling

### LINKED ISSUE #3147: [Feature Request] Question about LSTMModules
## Motivation

Could I get more details on tensor sizes required when using LSTMModule? 

I'm particularly confused about the required input size with recurrent mode... seems like it needs to be (*b, T, Feature), and that the time dimension *should not* have consecutive trajectories stacked. And since recurrent_mode is enabled for all losses, doesn't this mean every time you call a loss function and happen to have a recurrent model, it needs to be batched in the same way?

I'm a bit confused how this example works at all: https://docs.pytorch.org/rl/main/tutorials/dqn_with_rnn.html. The data passed into the loss function is of size (B1, B2), where both batch dimensions have multiple slices concatentated together. Wouldn't this mean that during training the recurrent state is being passed between unrelated trajectories?

In [2]: s["collector", "traj_ids"]
Out[2]: 
tensor([[10, 10, 10, 10, 10, 10, 10, 10, 11, 11, 11, 11, 11, 11, 11, 11, 11, 12,
         12, 12, 12, 12, 12, 12, 12, 12, 13, 13, 13, 13, 13, 13, 13, 13, 13, 13,
         13, 13, 13, 13, 13, 13, 14, 14, 14, 14, 14, 14, 14, 14],
        [30, 30, 30, 30, 30, 30, 30, 30, 30, 30, 30, 31, 31, 31, 31, 31, 31, 31,
         31, 32, 32, 32, 32, 32, 32, 32, 32, 33, 33, 33, 33, 33, 33, 33, 33, 33,
         33, 33, 33, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34],
        [20, 20, 20, 20, 20, 20, 20, 21, 21, 21, 21, 21, 21, 21, 21, 21, 22, 22,
         22, 22, 22, 22, 22, 22, 22, 22, 23, 23, 23, 23, 23, 23, 23, 23, 23, 23,
         23, 23, 24, 24, 24, 24, 24, 24, 24, 24, 24, 24, 25, 25],
        [ 0,  0,  0,  0,  0,  0,  0,  0,  0,  1,  1,  1,  1,  1,  1,  1,  1,  1,
          1,  1,  2,  2,  2,  2,  2,  2,  2,  2,  2,  2,  3,  3,  3,  3,  3,  3,
          3,  3,  3,  3,  4,  4,  4,  4,  4,  4,  4,  4,  4,  4]])

## Solution

Documentation updates. 

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by YeonwooSung**
## Description

Document the `is_init` contract after slice sampling, and add a CPU regression that hidden state leaks across concatenated mid-episode slices when that contract is skipped.

`InitTracker` only marks episode starts. A `SliceSampler` window can start mid-episode, so every stored `is_init` flag in that window is `False`. Concatenating those slices looks like one long trajectory to `LSTMModule`: hidden state from the last step of slice A is fed into the first step of unrelated slice B.

The contract: after slice sampling, mark the first timestep of each slice as `is_init=True`. `SliceSampler` already does this by default (`init_key="is_init"`). Hand-built batches can OR the first timestep of each slice, or shift `("next", "truncated")` onto `is_init` (do not OR `truncated` in place; it sits at slice ends). Sequential mode then zeros the incoming hidden with `torch.where(is_init, zeros, hidden)`; recurrent mode splits on `is_init` and restarts from the stored hidden at each split.

No LSTM internals are changed.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

close #3147

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTING](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) and [Agents.md](https://github.com/pytorch/rl/blob/main/Agents.md) docs
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have updated the documentation related to my changes

### CODE REVIEW

**Comment by itwasabhi** [comment]
Hi @YeonwooSung thanks for picking up this request! The docs are quite helpful.
